src/posit/connect/permissions.py

```diff
@@ -2,16 +2,27 @@
 
 from __future__ import annotations
 
-from typing import List, overload
+from typing import TYPE_CHECKING, List, overload
 
 from requests.sessions import Session as Session
 
+from ._deprecated import warn_deprecated_and_removed_in_future
 from .resources import Resource, ResourceParameters, Resources
 
+if TYPE_CHECKING:
+    from .groups import Group
+    from .users import User
+
 
 class Permission(Resource):
     def delete(self) -> None:
-        """Delete the permission."""
+        """[Deprecated] Delete the permission."""
+        warn_deprecated_and_removed_in_future("Please use .destroy() instead of .delete().")
```

Comment:
We aren't at 1.0 yet, so do we need to be so cautious about deprecation cycles?

Reply:
My thought was that the deprecations could be gutted when v1 comes out.

Reply:
I don't feel strongly, just that we probably have only a handful of users, if any, of the existing code, so it's probably not worth much trouble. Also, for recipes running on Connect, I don't think we meaningfully expose DeprecationWarnings to the owners (they probably show up in the logs, but you'd have to know to look for them), so the practical value is limited unless/until that is surfaced better.

Reply:
Great! I'll remove the methods and deprecation codes. 😁😁